content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Manual Rotation of CA Certificates
+min-kubernetes-server-version: v1.13
+content_template: templates/task
+---
+
+{{% capture overview %}}
+
+This page shows how to manually rotate the certificate authority (CA) certificates.
+
+{{% /capture %}}
+
+
+{{% capture prerequisites %}}
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+For more information about authentication in Kubernetes, see [Authenticating](/docs/reference/access-authn-authz/authentication).
+For more information about best practices for CA certificates, see [Single root CA](docs/setup/best-practices/certificates/#single-root-ca).
+
+{{% /capture %}}
+
+{{% capture steps %}}
+## Choose an approach
+There are two ways to rotate or replace CA certificates.
+
+## Replace the cluster (recommended) {#replace-the-cluster}
+
+This is the safest approach to rotate or replace the CA certificates.
+
+1. Create a new cluster with new CA certificates.
+
+2. Migrate the workloads to the new cluster.
+
+3. Remove or destroy the old cluster.
+
+## Rotate the CA certificates manually
+
+{{< caution >}}
+Make sure to back up your certificate directory along with configuration files (`.conf`) and any other necessary files.
+
+Manually rotating the CA works by timing when a Kubernetes cluster knows about new certificates. Do not restart any pods or update any component in the cluster while manually rotating the CA.
+
+This approach requires a considerable downtime of your cluster.
+
+{{< /caution >}}
+    
+1. Distribute new CA private keys (`ca.key` and `front-proxy-ca.key`) to all your control plane nodes in the Kubernetes certificates directory.
+
+1. Append new CA to the existing CA files (`ca.crt` and `front-proxy-ca.crt`) and the file against `--client-ca-file` flag in the `kube-apiserver` and `kube-scheduler` configuration.
+
+    {{< note >}}
+
+    If you used `kubeadm` to deploy your cluster, the control plane components run as {{< glossary_tooltip text="static Pods" term_id="static-pod" >}}, managed by the {{< glossary_tooltip text="kubelet" term_id="kubelet" >}}. Components are not disrupted by in-place updates of CA certificates until someone kills or restarts an already running Pod.
+
+    {{< /note >}}
+
+1. Update *Kubernetes controller manager's* `--root-ca-file` to include both old and new CA.
+
+     {{< note >}}
+
+     Remove the flag `--client-ca-file` from the *Kubernetes controller manager* configuration. You can also replace the existing client CA file or change this configuration item to reference a new, updated CA. [Issue 1350](https://github.com/kubernetes/kubeadm/issues/1350) enumerates an issue with *Kubernetes controller manager* being unable to accept CA bundle.
+
+    {{< /note >}}
+
+1. Update the kubelet configuration to use new CA on all worker nodes.
```

Comment:
`use the new`

Reply:
i think some more detail here is needed.

is this about updating `clientCAFile: ` in the KubeletConfiguration (e.g. `/var/lib/kubelet/config.yaml`) to point to a new CA file? default value might be `clientCAFile: /etc/kubernetes/pki/ca.crt`.

yet earlier we recommended that the new CA is appended to the `ca.crt`.

Reply:
Q: why only workers? can this be done for all nodes?

Reply:
This should call out that its the clientCA file _and_ the kubeconfig on all nodes (not just worker)

Reply:
agreed @micahhausler @neolit123 , my bad. Will update